Failure to compile on MacOS (metal crate)

Hi! I'm really excited to try out nannou. I'm having some issues compiling it on macOS 10.14.5. When I run `cargo run --release --example simple_draw` in the nannou repository, I'm getting an error compiling the `metal` crate.

I've tried compiling on both of these toolchains:
```
stable-x86_64-apple-darwin -- rustc 1.38.0 (625451e37 2019-09-23)
nightly-x86_64-apple-darwin -- rustc 1.40.0-nightly (c23a7aa77 2019-10-19)
```

On both toolchains, I get an error compiling the `metal` crate:

```
Compiling metal v0.13.1
error[E0282]: type annotations needed
   --> /Users/bencongdon/.cargo/registry/src/github.com-1ecc6299db9ec823/metal-0.13.1/src/lib.rs:128:13
    |
128 |             msg_send![self.0, release];
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ consider giving `result` a type
    |
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

Here's the full error (though I don't think it yields much more information): https://pastebin.com/9YXijsjZ

Do you have any ideas of how I could fix this? Thanks! 😄

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,7 @@ addons:
     apt:
         packages:
             - libasound2-dev
+            - libxcb-composite0-dev
 matrix:
     include:
         - name: "Test nightly"

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,17 @@
 # Unreleased
 
+# Version 0.12.0 (2019-11-03)
 
+- Update vulkano dependencies to 0.16 in order to address `metal` related bug on
+  macOS.
+- Update conrod dependencies to 0.68 for vulkano patch. New version includes
+  copy/paste, double-click select and shift-click select support for the
+  `TextEdit` widget.
+- [Breaking] Small change to Vulkan debug items.
+- [Breaking] New fields have been added to `DynamicState`.
+- Update shade_runner to 0.3 for vulkano patch.
+- Frame command buffer builder no longer provides access to unrelated
+  `secondary` buffer methods.
 
 # Version 0.11.0 (2019-09-17)
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,7 +44,7 @@ audrey = "0.2"
 nannou_audio = "0.2"
 nannou_laser = "0.3"
 nannou_osc = "0.1"
-shade_runner = { git = "https://github.com/mitchmindtree/shade_runner", branch = "update_vulkano" }
+shade_runner = "0.3"
 
 [target.'cfg(target_os = "macos")'.dependencies]
 moltenvk_deps = "0.1"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "nannou"
-version = "0.11.1"
+version = "0.12.0"
 authors = ["mitchmindtree <mitchell.nordine@gmail.com>"]
 description = "A Creative Coding Framework for Rust."
 readme = "README.md"
@@ -16,9 +16,9 @@ default = ["notosans"]
 [dependencies]
 approx = "0.1"
 cgmath = { version = "0.16", features = ["serde"] }
-conrod_core = "0.67"
-conrod_winit = "0.67"
-conrod_vulkano = "0.67"
+conrod_core = "0.68"
+conrod_winit = "0.68"
+conrod_vulkano = "0.68"
 daggy = "0.6"
 find_folder = "0.3"
 image = "0.21"
@@ -33,9 +33,9 @@ serde = "1"
 serde_derive = "1"
 serde_json = "1"
 toml = "0.4"
-vulkano = "0.14"
-vulkano-win = "0.14"
-vulkano-shaders = "0.14"
+vulkano = "0.16"
+vulkano-win = "0.16"
+vulkano-shaders = "0.16"
 walkdir = "2"
 winit = "0.19"
 
@@ -44,7 +44,7 @@ audrey = "0.2"
 nannou_audio = "0.2"
 nannou_laser = "0.3"
 nannou_osc = "0.1"
-shade_runner = "0.2"
+shade_runner = { git = "https://github.com/mitchmindtree/shade_runner", branch = "update_vulkano" }
 
 [target.'cfg(target_os = "macos")'.dependencies]
 moltenvk_deps = "0.1"

--- a/src/app.rs
+++ b/src/app.rs
@@ -554,7 +554,7 @@ where
 
                 vk_builder
                     .add_extensions(vk::InstanceExtensions {
-                        ext_debug_report: true,
+                        ext_debug_utils: true,
                         ..vk::InstanceExtensions::none()
                     })
                     .layers(vec!["VK_LAYER_LUNARG_standard_validation"])

--- a/src/vk.rs
+++ b/src/vk.rs
@@ -120,7 +120,7 @@ pub use vulkano_win as win;
 
 use crate::vk;
 use crate::vk::instance::debug::{
-    DebugCallback, DebugCallbackCreationError, Message, MessageTypes,
+    DebugCallback, DebugCallbackCreationError, Message, MessageSeverity, MessageType,
 };
 use crate::vk::instance::loader::{FunctionPointers, Loader};
 use std::borrow::Cow;
@@ -177,7 +177,8 @@ pub struct InstanceBuilder {
 /// A builder struct that makes the process of building a debug callback more modular.
 #[derive(Default)]
 pub struct DebugCallbackBuilder {
-    pub message_types: Option<MessageTypes>,
+    pub message_type: Option<MessageType>,
+    pub message_severity: Option<MessageSeverity>,
     pub user_callback: Option<BoxedUserCallback>,
 }
 
@@ -330,7 +331,7 @@ impl InstanceBuilder {
                 e.khr_wayland_surface |= ext.khr_wayland_surface;
                 e.khr_android_surface |= ext.khr_android_surface;
                 e.khr_win32_surface |= ext.khr_win32_surface;
-                e.ext_debug_report |= ext.ext_debug_report;
+                e.ext_debug_utils |= ext.ext_debug_utils;
                 e.mvk_ios_surface |= ext.mvk_ios_surface;
                 e.mvk_macos_surface |= ext.mvk_macos_surface;
                 e.mvk_moltenvk |= ext.mvk_moltenvk;
@@ -394,11 +395,19 @@ impl DebugCallbackBuilder {
         Default::default()
     }
 
+    /// The severity of messages emitted via the debug callback.
+    ///
+    /// If unspecified, nannou will use `MessageType::errors_and_warnings`.
+    pub fn message_severity(mut self, msg_sev: MessageSeverity) -> Self {
+        self.message_severity = Some(msg_sev);
+        self
+    }
+
     /// The message types to be emitted to the debug callback.
     ///
-    /// If unspecified, nannou will use `MessageTypes::errors_and_warnings`.
-    pub fn message_types(mut self, msg_tys: MessageTypes) -> Self {
-        self.message_types = Some(msg_tys);
+    /// If unspecified, nannou will use `MessageType::errors_and_warnings`.
+    pub fn message_type(mut self, msg_ty: MessageType) -> Self {
+        self.message_type = Some(msg_ty);
         self
     }
 
@@ -419,39 +428,43 @@ impl DebugCallbackBuilder {
         instance: &Arc<Instance>,
     ) -> Result<DebugCallback, DebugCallbackCreationError> {
         let DebugCallbackBuilder {
-            message_types,
+            message_severity,
+            message_type,
             user_callback,
         } = self;
-        let message_types = message_types.unwrap_or_else(|| MessageTypes {
-            error: true,
-            warning: true,
-            performance_warning: true,
-            information: true,
-            debug: true,
-        });
+        let message_severity = message_severity.unwrap_or_else(MessageSeverity::errors_and_warnings);
+        let message_type = message_type.unwrap_or_else(MessageType::all);
         let user_callback = move |msg: &Message| {
             match user_callback {
                 Some(ref cb) => (**cb)(msg),
                 None => {
-                    let ty = if msg.ty.error {
+                    let sev = if msg.severity.error {
                         "error"
-                    } else if msg.ty.warning {
+                    } else if msg.severity.warning {
                         "warning"
-                    } else if msg.ty.performance_warning {
-                        "performance_warning"
-                    } else if msg.ty.information {
+                    } else if msg.severity.information {
                         "information"
-                    } else if msg.ty.debug {
-                        "debug"
+                    } else if msg.severity.verbose {
+                        "verbose"
                     } else {
-                        println!("[vulkan] <unknown message type>");
-                        return;
+                        "<unknown severity>"
                     };
-                    println!("[vulkan] {} {}: {}", msg.layer_prefix, ty, msg.description);
+
+                    let ty = if msg.ty.general {
+                        "general"
+                    } else if msg.ty.validation {
+                        "validation"
+                    } else if msg.ty.performance {
+                        "performance"
+                    } else {
+                        "unknown type"
+                    };
+
+                    println!("[vulkan] {} {} {}: {}", msg.layer_prefix, ty, sev, msg.description);
                 }
             };
         };
-        DebugCallback::new(instance, message_types, user_callback)
+        DebugCallback::new(instance, message_severity, message_type, user_callback)
     }
 }
 

--- a/src/vk.rs
+++ b/src/vk.rs
@@ -432,7 +432,8 @@ impl DebugCallbackBuilder {
             message_type,
             user_callback,
         } = self;
-        let message_severity = message_severity.unwrap_or_else(MessageSeverity::errors_and_warnings);
+        let message_severity =
+            message_severity.unwrap_or_else(MessageSeverity::errors_and_warnings);
         let message_type = message_type.unwrap_or_else(MessageType::all);
         let user_callback = move |msg: &Message| {
             match user_callback {
@@ -460,7 +461,10 @@ impl DebugCallbackBuilder {
                         "unknown type"
                     };
 
-                    println!("[vulkan] {} {} {}: {}", msg.layer_prefix, ty, sev, msg.description);
+                    println!(
+                        "[vulkan] {} {} {}: {}",
+                        msg.layer_prefix, ty, sev, msg.description
+                    );
                 }
             };
         };

